### PR TITLE
pg_authid is not avaialble on aws rds postgres

### LINKED ISSUE #169: community.postgresql.postgresql_info does not work in AWS RDS hosted postgresql
the table `pg_authid` is not available on AWS RDS hosted postgresql.

It just gives the following error: `PostgreSQL said: permission denied for relation pg_authid`

The problem is that in the AWS managed RDS, once can't have the full superuser, just some aws admin user that can't access this table.

It seems that AWS consides this a feature, not a bug, source: https://stackoverflow.com/questions/55143544/aws-rds-postgresql-access-to-pg-catalog-pg-authid-forbidden-in-all-contexts

Since `postgres_info` uses a query that involves this table: https://github.com/ansible-collections/community.postgresql/blob/main/plugins/modules/postgresql_info.py#L711

that command does not work on aws rds.

### LINKED ISSUE #170: Make this collection AWS-compatible
##### SUMMARY
Reconstruct some queries to bypass certain PG system tables like **pg_authid**, that are not accessible in AWS.  We use the **pg_authid**  table a lot for getting the name associated with an owner OID number.

##### ISSUE TYPE
- Make this collection AWS compatible.

##### COMPONENT NAME
Right now, I have identified the **postgresql_info.py** module component as where SQL changes are needed.

##### ADDITIONAL INFORMATION
For instance, I documented a fix in issue, https://github.com/ansible-collections/community.postgresql/issues/169.
So for the case where the SQL is this:
`SELECT s.spcname, a.rolname, s.spcacl FROM pg_tablespace AS s JOIN pg_authid AS a ON s.spcowner = a.oid;`
Change it to this:
`SELECT spcname, pg_catalog.pg_get_userbyid(spcowner) AS rolname, spcacl FROM pg_catalog.pg_tablespace;`

### PR DESCRIPTION
**Pull request by marcosdiez**
closes #170
closes #169 

@MichaelDBA please code review if you may. Thank you for your help!

### CODE REVIEW

File: plugins/modules/postgresql_info.py [hunk 1]
```diff
@@ -935,9 +931,8 @@ def get_lang_info(self):
 
     def get_namespaces(self):
         """Get information about namespaces."""
-        query = ("SELECT n.nspname, a.rolname, n.nspacl "
-                 "FROM pg_catalog.pg_namespace AS n "
-                 "JOIN pg_authid AS a ON a.oid = n.nspowner")
+        query = ("SELECT n.nspname, pg_catalog.pg_get_userbyid(a.oid) AS rolname, n.nspacl "
+                 "FROM pg_catalog.pg_namespace AS n ")
```
Comment (tcraxs) [inline plugins/modules/postgresql_info.py:935]: > missing FROM-clause entry for table "a"

`a.oid` is not possible here

> [testhost] testhost: postgresql_info : postgresql_info - test return values and session_role param login_user={{ pg_user }}, login_db={{ db_default }}, login_port={{ replica_port }}, session_role=session_superuser
> Cannot execute SQL 'SELECT n.nspname, pg_catalog.pg_get_userbyid(a.oid) AS rolname, n.nspacl FROM pg_catalog.pg_namespace AS n ': missing FROM-clause entry for table "a" LINE 1: SELECT n.nspname, pg_catalog.pg_get_userbyid(a.oid) AS rolna...                                                      ^

**Review by tcraxs — CHANGES_REQUESTED** [review]

File: plugins/modules/postgresql_info.py (hunk 1 above)
Comment (marcosdiez) [reply plugins/modules/postgresql_info.py]: my bad. fixed!

**Comment by tcraxs** [comment]
@marcosdiez Thanks for fixing the CI problem.
Could you also add a changelog fragment for this PR: https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment

Could you also check if integration tests are needed for this change. See the content of tests/integration/target/postgresql_info/tasks

**Comment by marcosdiez** [comment]
Hi @tcraxs 

This PR specifically does not need unit tests (or the exiting ones suffice)
I just added the  changelog fragment file.

now, somehow my tests are failing and I don't know why.
please help me

**Review by MichaelDBA — APPROVED** [review]
I was able to validate all the sql changes in this python module.

**Comment by Andersson007** [comment]
closed reopened to relaunch CI

**Review by Andersson007 — APPROVED** [review]
LGTM

**Comment by Andersson007** [comment]
@marcosdiez thank you for the contribution! (CI errors weren't related to you fix). Great job!
@tcraxs @MichaelDBA thank you for reviewing and testing!

**Pull request merged**